### PR TITLE
[E2S 1.0] Xarray model protocols and execution

### PR DESCRIPTION
**Pull request by NickGeneva**
## Stack
Stacked on #1167 (coordinate-system update). Base: `ngeneva/coordinate-systems`.

## Summary
- Update prognostic and diagnostic protocols to DataArray inputs/outputs and migrate FCN and PrecipitationAFNO execution.
- Add shared DataArray call/generator batching through `.e2s.batch()` / `.e2s.unbatch()`, with coordinate restoration and batch-order validation.
- Extend the Torch bridge to accept coordinate signatures; preserve FCN iterator hooks, six-hour stepping, and checkpoint metadata including temporal statistics.
- Add execution tests and a runnable tutorial; document the incremental migration boundary and legacy conformance checker scope.

## Validation
- Focused model, batching, coordinate, and bridge suite: **89 passed, 30 skipped**; skips cover missing CuPy and optional model extras.
- Black and Ruff passed on all 14 changed Python files.
- CPU execution tutorial and `git diff --check` passed.
- Focused mypy passed with missing imports ignored (PhysicsNeMo AFNO is unavailable in the environment).

## Migration scope
FCN and PrecipitationAFNO use the new one-DataArray execution API. Other models and inference drivers remain on their existing execution APIs; Random and Random_FX are unchanged. The existing conformance checker remains scoped to legacy tensor models.

### CODE REVIEW

**Review by pzharrington — APPROVED** [review]

File: earth2studio/models/px/fcn.py [hunk 1]
```diff
@@ -250,78 +252,60 @@ def _forward(self, x: torch.Tensor) -> torch.Tensor:
...
-
-            yield x, coords.copy()
+            x = self.rear_hook(self._step(self.front_hook(x)))
+            self._save_checkpoint_state(x)
+            yield x.copy(deep=False)
 
-    def create_iterator(
-        self, x: torch.Tensor, coords: CoordSystem
-    ) -> Iterator[tuple[torch.Tensor, CoordSystem]]:
+    def create_iterator(self, x: xr.DataArray) -> Iterator[xr.DataArray]:
```
[the remark this answers is not in the document]
Comment (NickGeneva) [reply earth2studio/models/px/fcn.py:296]: Doesnt matter
